File: csharp/0235-lowest-common-ancestor-of-a-binary-search-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */

public class Solution {
    public TreeNode LowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        // Traverse Right child
        if (p.val > root.val && q.val > root.val) {
            return LowestCommonAncestor(root.right, p, q);
        }
        
        // Traverse Left Child
        if (p.val < root.val && q.val < root.val) {
            return LowestCommonAncestor(root.left, p, q);
        }
        
        return root;
    }
}

File: c/0235-lowest-common-ancestor-of-a-binary-search-tree.c

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

struct TreeNode* lowestCommonAncestor(struct TreeNode* root, struct TreeNode* p, struct TreeNode* q) {
    
    int rootVal = root -> val;
    int pVal = p -> val;
    int qVal = q -> val;

    if (pVal > rootVal && qVal > rootVal) {
        return lowestCommonAncestor(root -> right, p, q);
    } else if (pVal < rootVal && qVal < rootVal) {
        return lowestCommonAncestor(root -> left, p, q);
    }
    return root;
}